drivers/timer/hi3861_timer.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT hisilicon_hi3861_timer
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/irq.h>
+
+#include <hisilicon/hi3861/timer.h>
+
+#define TIMER_REG DT_INST_REG_ADDR(0)
+
+#define TIMER_0 ((HI3861_TIMER *)(TIMER_REG + 0x00))
+#define TIMER_1 ((HI3861_TIMER *)(TIMER_REG + 0x14))
+#define TIMER_2 ((HI3861_TIMER *)(TIMER_REG + 0x28))
+#define TIMER_3 ((HI3861_TIMER *)(TIMER_REG + 0x3C))
+
+#define CYCLES_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static volatile uint64_t last_ticks = 0;
+
+static void sys_clock_isr(const void *arg)
+{
+	ARG_UNUSED(arg);
+
+	// Read EOI to clear interrupt
+	uint32_t n = TIMER_3->EOI;
```

Comment:
please use the helper `sys_read32` instead (same for all the other occurrences).

Reply:
Are you suggesting that accessing registers using the `BASE + OFFSET` method is better than accessing struct members by casting a pointer to the base address?

Reply:
definitely. With pointer and base address you are offloading the task of creating the correct code to the compiler, using the helpers you are avoiding that. Also the helpers sometimes add some kind of optimizations, like memory barriers, etc...